lib/repl.js

```diff
@@ -1119,6 +1173,18 @@ function defineDefaultCommands(repl) {
       this.displayPrompt();
     }
   });
+
+  repl.defineCommand('paste', {
+    help: 'Entering paste mode (^D to finish, ^C to cancel)',
+    action: function() {
+      if (this.terminal) {
+        this.pasteMode = true;
+        REPLServer.super_.prototype.setPrompt.call(this, '');
```

Comment:
@Fishrock123 I am preserving `this.__initialPrompt` and temporarily setting empty string to `readline`'s prompt. Its needed for various terminal shortcuts which will pollute the paste mode.